### ISSUE TITLE
Segfault when changing the filed names of an structured array (migrated from Trac #1598)

### ISSUE THREAD

**Issue by thouis** [thouis/numpy-trac-migration#3149]
Original ticket http://projects.scipy.org/numpy/ticket/1598
Reported 2010-09-01 by atmention:FrancescAlted, assigned to unknown.

The next reproduces the segfault:

```
In [1]: import numpy as np

In [2]: np.__version__
Out[2]: '1.5.0'

In [3]: ra = np.fromiter(((i*3, i*2) for i in xrange(10)), dtype='i8,f8')

In [4]: ra.dtype.names = ('f1', 'f2')

In [5]: ra
Segmentation fault
```

I don't know if the `names` filed of a dtype can be changed like this, but numpy let me do that, so...

**Comment by thouis** [comment]
Comment in Trac by atmention:pierregm, 2010-09-13

Confirmed, but it's a bit more complex than that:
    >>>  a=np.array(zip((1,2,3),(1,2,3)),dtype=[('a',int),('b',float)])
    >>> a.dtype.names = ('a','c')
    >>> a
    array([(1, 1.0), (2, 2.0), (3, 3.0)], 
          dtype=[('a', '<i8'), ('c', '<f8')])
So, it looks like the segfault shows up when renaming a field by the name of another existing field.

**Comment by thouis** [comment]
Comment in Trac by atmention:pv, 2010-10-10

Fixed in commit:a16ee31 and commit:462830b
